### DISCUSSION TITLE
How replace TriStateCheckbox by Checkbox ?

### DISCUSSION THREAD

**Discussion by Quent1L** [primefaces/community#2463] [PrimeVue]
Hello,
Since the release of version 4, the TriStateCheckbox component has been removed and the indeterminate property has been added to the Checkbox component. Except that I can't get the same functionality as the TriStateCheckbox before.

https://stackblitz.com/edit/ghfhbr
Is it possible to simplify this? Have I misunderstood the use of indeterminate props?

Thank you for your answers.

**Comment by j0rgedev** [comment]
Hello @Quent1L. What you did is a good workaround. However, I can't think of a way to simplify it. In V3 it was already built-in but now you need to handle it. Maybe, it was a decision to have more control over it.

**Comment by prabinsh** [comment]
@Quent1L take a look at this thread #2957
